lib/marked.js

```diff
@@ -159,49 +159,70 @@ Lexer.prototype.token = function(src, top, bq) {
     , i
     , l;
 
+  var lineNumber = this.options.lineNumber;
+  var lines = lineNumber ? 
+    function(content) {
+      return content.split('\n').length - 1;
+    } : 
+    function() { return 0; };
+
+  var tokens = this.tokens;
+  var push = function(token) {
+    if (lineNumber) {
+      token.ln = ln;
+    }
+    tokens.push(token);
+  };
+
```

Comment:
instead of defining a new push function, override  `this.tokens.push()` seems more graceful:

```
var originPush = this.tokens.push;
this.tokens.push = function (token) {
  if (lineNumber) {
     token.ln = ln;
  }
 originPush.call(this, token);
};
```

otherwise test case is necessary